### ISSUE TITLE
Keyboard Shortcut Regression: Option+Delete Incorrectly Deletes Multiple Words in Korean Input

### ISSUE THREAD

**Issue by dompoo** [anthropics/claude-code#2742]
**Bug Description**
When I use korean, option+delete doesn't work well. It deletes many words at once. PLEASE FIX THIS!!!

**Environment Info**
- Platform: darwin
- Terminal: intellij
- Version: 1.0.35
- Feedback ID: 547d70f1-18d7-44eb-aa80-7aaf59b41120

**Comment by Kolger** [comment]
#1943
